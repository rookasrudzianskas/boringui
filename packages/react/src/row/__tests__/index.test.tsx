import React from "react";
import {mount} from "enzyme";

import {Row, Col} from "../../index";

describe("Row", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Row>row</Row>);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should be render different components", () => {
    const wrapper = mount(<Row>row</Row>);

    wrapper.setProps({as: "div"});
    expect(wrapper.find("div").length).not.toBe(0);

    wrapper.setProps({as: "nav"});
    expect(wrapper.find("nav").length).not.toBe(0);

    wrapper.setProps({as: "span"});
    expect(wrapper.find("span").length).not.toBe(0);

    wrapper.setProps({as: "p"});
    expect(wrapper.find("p").length).not.toBe(0);

    wrapper.setProps({as: "details"});
    expect(wrapper.find("details").length).not.toBe(0);
  });

  it("the children should be aligned correctly", () => {
    const wrapper = mount(
      <div>
        <Row justify="flex-end">
          <Col />
        </Row>
        <Row justify="center">
          <Col />
        </Row>
        <Row justify="space-around">
          <Col />
        </Row>
        <Row justify="space-between">
          <Col />
        </Row>
        <Row align="flex-start">
          <Col />
        </Row>
        <Row align="center">
          <Col />
        </Row>
        <Row align="flex-end">
          <Col />
        </Row>
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("the children should have the correct spacing", () => {
    const wrapper = mount(
      <div>
        <Row gap={1}>
          <Col />
        </Row>
        <Row gap={2}>
          <Col />
        </Row>
        <Row gap={10}>
          <Col />
        </Row>
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with nested", () => {
    const wrapper = mount(
      <Row>
        <Row>
          <Row>
            <Col>
              <Row>row</Row>
            </Col>
          </Row>
        </Row>
      </Row>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });
});
