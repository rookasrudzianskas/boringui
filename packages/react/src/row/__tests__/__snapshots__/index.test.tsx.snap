// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `Row should render correctly 1`
] = `"<div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\">row</div>"`;

exports[
  `Row should work with nested 1`
] = `"<div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\"><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\"><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\">row</div></div></div></div></div>"`;

exports[
  `Row the children should be aligned correctly 1`
] = `"<div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-iipPaye-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-iiPCojg-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-icEZgMd-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-igmUkpg-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idIWWzx-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ilailYI-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ibEUQES-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div></div>"`;

exports[
  `Row the children should have the correct spacing 1`
] = `"<div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ieLOVho-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-igSdzKj-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div><div class=\\"nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ifliOBE-css\\"><div class=\\"nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css\\"></div></div></div>"`;
